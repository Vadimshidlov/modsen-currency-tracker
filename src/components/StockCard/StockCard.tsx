import React from "react";
import Text from "@/components/Text/Text";
import "@/components/CurrencyCard/CurrencyCard.scss";
import BovespaIcon from "@/assets/svg/stocks/bovespa-icon.svg";
import IfixIcon from "@/assets/svg/stocks/ifix-icon.svg";

export type CurrencyPropsType = {
    stockCode: string;
    value: number;
};

export default function StockCard({ stockCode, value }: CurrencyPropsType) {
    let CurrencyIcon;
    let currencyTitle = "";
    const currencyValue = +value.toFixed(2);

    switch (stockCode) {
        case "BOVESPA_INDEX":
            CurrencyIcon = BovespaIcon;
            currencyTitle = "Bovespa Index";
            break;
        case "IIFIX":
            CurrencyIcon = IfixIcon;
            currencyTitle = "IFIX";
            break;

        default:
            CurrencyIcon = BovespaIcon;
            currencyTitle = "Bovespa Index";
    }

    return (
        <button className="currency-card__container" disabled>
            <CurrencyIcon className="currency-card__icon" />
            <div className="currency-card__info">
                <Text className="currency-card__titile">{currencyTitle}</Text>
                <Text className="currency-card__percent">{`${currencyValue}%`}</Text>
            </div>
        </button>
    );
}
