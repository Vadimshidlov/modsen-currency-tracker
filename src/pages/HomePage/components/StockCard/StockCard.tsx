import React from "react";
import Text from "@/pages/HomePage/components/Text/index";
import "@/pages/HomePage/components/CurrencyCard/CurrencyCard.scss";
import BovespaIcon from "@/assets/svg/stocks/bovespa-icon.svg";
import IfixIcon from "@/assets/svg/stocks/ifix-icon.svg";
import { CurrencyStockPropsType } from "@/types/HomePageTypes";

export default function StockCard({ stockCode, value }: CurrencyStockPropsType) {
    let CurrencyIcon;
    let currencyTitle = "";
    const currencyValue = +value.toFixed(2);

    switch (stockCode) {
        case "BOVESPA_INDEX":
            CurrencyIcon = BovespaIcon;
            currencyTitle = "Bovespa Index";
            break;
        case "IIFIX":
            CurrencyIcon = IfixIcon;
            currencyTitle = "IFIX";
            break;

        default:
            CurrencyIcon = BovespaIcon;
            currencyTitle = "Bovespa Index";
    }

    return (
        <button className="currency-card__container" disabled>
            <CurrencyIcon className="currency-card__icon" />
            <div className="currency-card__info">
                <Text className="currency-card__titile">{currencyTitle}</Text>
                <Text className="currency-card__percent">{`${currencyValue}%`}</Text>
            </div>
        </button>
    );
}
