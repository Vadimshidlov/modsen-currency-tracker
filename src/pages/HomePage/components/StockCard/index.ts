export { default } from "@/pages/HomePage/components/StockCard/StockCard";
