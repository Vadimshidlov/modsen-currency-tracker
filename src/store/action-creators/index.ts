export { getMapData } from "@/store/action-creators/getMapData";
export { getCurrency } from "@/store/action-creators/getCurrency";
export {
    selectSecondCurrency,
    closeModalWindow,
    openModalWindow,
} from "@/store/action-creators/openModalWindow";
export { switchLightTheme, switchDarkTheme } from "@/store/action-creators/toggleTheme";
export {
    errorAuthUserLocation,
    successAuthUserLocation,
} from "@/store/action-creators/authUserLocation";
