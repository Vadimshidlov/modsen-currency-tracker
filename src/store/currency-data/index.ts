export { mockMapData } from "@/store/currency-data/mockMapData";
export { mockBanksApiData } from "@/store/currency-data/mockBanksApiData";
export { mockStockData } from "@/store/currency-data/mockStockData";
