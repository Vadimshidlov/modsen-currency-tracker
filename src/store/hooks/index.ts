export { useActions } from "@/store/hooks/useActions";
export { useTypedSelectorHook } from "@/store/hooks/useTypedSelector";
