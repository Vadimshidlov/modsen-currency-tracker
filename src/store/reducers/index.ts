import { combineReducers } from "redux";
import themeReducer from "@/store/reducers/themeReducer";
import latestCurrencyReducer from "@/store/reducers/latestCurrencyReducer";
import modalWindowReducer from "@/store/reducers/modalWindowReducer";
import mapDataReducer from "@/store/reducers/mapDataReducer";

export const rootReducer = combineReducers({
    theme: themeReducer,
    currency: latestCurrencyReducer,
    modalWindow: modalWindowReducer,
    mapData: mapDataReducer,
});

export type RootStateType = ReturnType<typeof rootReducer>;
