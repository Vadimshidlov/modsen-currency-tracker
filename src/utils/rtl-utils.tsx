import React, { PropsWithChildren } from "react";
import { Store, applyMiddleware, createStore } from "redux";
import { thunk } from "redux-thunk";
import { render } from "@testing-library/react";
import type { RenderOptions } from "@testing-library/react";
import { Provider } from "react-redux";
import { RootStateType, rootReducer } from "@/store/reducers";

interface ExtendedRenderOptions extends Omit<RenderOptions, "queries"> {
    preloadedState?: Partial<RootStateType>;
    store?: Store;
}

export function renderWithProviders(
    ui: React.ReactElement,
    {
        preloadedState = {},
        store = createStore(rootReducer, preloadedState, applyMiddleware(thunk)),
        ...renderOptions
    }: ExtendedRenderOptions = {},
) {
    function Wrapper({ children }: PropsWithChildren<object>): JSX.Element {
        return <Provider store={store}>{children}</Provider>;
    }
    return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) };
}

export * from "@testing-library/react";

export { renderWithProviders as render };
