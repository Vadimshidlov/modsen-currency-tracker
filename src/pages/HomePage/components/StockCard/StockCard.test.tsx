import "@testing-library/jest-dom";
import renderer from "react-test-renderer";
import { Provider } from "react-redux";
import { store } from "@/store";
import { renderWithProviders, screen } from "@/utils/rtl-utils";
import StockCard from "@/pages/HomePage/components/StockCard/index";

describe("CurrencyStocks component tests", () => {
    it("should correct render component", async () => {
        renderWithProviders(<StockCard stockCode="IIFX" value={0.15} />);

        expect(screen.getByText("Bovespa Index")).toBeInTheDocument();
        expect(screen.getByText("0.15%")).toBeInTheDocument();
    });

    it("snapshot matches", async () => {
        const { asFragment } = renderWithProviders(<StockCard stockCode="IIFX" value={0.15} />);
        expect(asFragment()).toMatchSnapshot();
    });

    it("should match snapshot", () => {
        const tree = renderer
            .create(
                <Provider store={store}>
                    <StockCard stockCode="IIFX" value={0.15} />
                </Provider>,
            )
            .toJSON();

        expect(tree).toMatchSnapshot();
    });
});
