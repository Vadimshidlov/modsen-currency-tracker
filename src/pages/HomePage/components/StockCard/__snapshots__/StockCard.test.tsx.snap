// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrencyStocks component tests should match snapshot 1`] = `
<button
  className="currency-card__container"
  disabled={true}
>
  <svg
    className="currency-card__icon"
  />
  <div
    className="currency-card__info"
  >
    <p
      className="currency-card__titile"
    >
      Bovespa Index
    </p>
    <p
      className="currency-card__percent"
    >
      0.15%
    </p>
  </div>
</button>
`;

exports[`CurrencyStocks component tests snapshot matches 1`] = `
<DocumentFragment>
  <button
    class="currency-card__container"
    disabled=""
  >
    <svg
      class="currency-card__icon"
    />
    <div
      class="currency-card__info"
    >
      <p
        class="currency-card__titile"
      >
        Bovespa Index
      </p>
      <p
        class="currency-card__percent"
      >
        0.15%
      </p>
    </div>
  </button>
</DocumentFragment>
`;
